rhino/src/main/java/org/mozilla/javascript/Context.java

```diff
@@ -1367,6 +1368,16 @@ public final Script compileReader(
      */
     public final Script compileReader(
             Reader in, String sourceName, int lineno, Object securityDomain) throws IOException {
+        return compileReader(in, sourceName, lineno, securityDomain, null);
+    }
+
+    public Script compileReader(
+            Reader in,
+            String sourceName,
+            int lineno,
+            Object securityDomain,
+            Consumer<CompilerEnvirons> compilerEnvironsProcessor)
```

Comment:
I don't think that we should be adding arguments to a public method that people definitely use outside this repo. It's probably fine to use a default argument, but I think it'd be safest to add a new method that has the new argument and keep the old one around for backward compatibility.

Reply:
Usually we mark the old as deprecated and taking care the old is not used any longer by ourself.

Reply:
We have done that! The old method is literally just above the line you commented on. We have also not deprecated it, because it felt it would annoy every single user to have to add a `, null` at the end or get a deprecation warning...